### PR TITLE
Add mecanum-bot example

### PR DESCRIPTION
**Pull request by fletch3555**
wpimath.kinematics.ChassisSpeeds.descretize(...) doesn't seem to exist.  Am I missing it somewhere? Is it new and needs to be added? Is there a different/better approach?

RobotPy docs: https://robotpy.readthedocs.io/projects/wpimath/en/latest/wpimath.kinematics/ChassisSpeeds.html
wpilibj source: https://github.com/wpilibsuite/allwpilib/blob/main/wpimath/src/main/java/edu/wpi/first/math/kinematics/ChassisSpeeds.java#L92

### CODE REVIEW

**Comment by fletch3555** [comment]
> It was added in the beta, and the docs were not updated yet. The doc site has moved, and you can see that function here: https://robotpy.readthedocs.io/projects/robotpy/en/latest/wpimath.kinematics/ChassisSpeeds.html#wpimath.kinematics.ChassisSpeeds.discretize

Good call.  I didn't realize I wasn't running the 2024 beta version.  Should be updated now

File: mecanum-bot/drivetrain.py [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+import math
+
+
+class Drivetrain:
+    """Represents a differential drive style drivetrain."""
+
+    MAX_SPEED = 3.0  # 3 meters per second
+    MAX_ANGULAR_SPEED = math.pi  # 1/2 rotation per second
+
+    # TRACK_WIDTH = 0.381 * 2  # meters
```
Comment (virtuald) [inline mecanum-bot/drivetrain.py:18]: Why are these comments here?

**Review by virtuald — CHANGES_REQUESTED** [review]
If you rebase on main and add the copyright statement I think we're good here.

File: mecanum-bot/drivetrain.py (hunk 1 above)
Comment (fletch3555) [reply mecanum-bot/drivetrain.py]: Just copy/paste... They're gone now.

**Pull request merged**